Update for the 10.3.0 release (#643)
bottlerocket-os/bottlerocket-core-kit: CHANGELOG.md

> # v10.3.0 (2025-08-25)

Reviewer: nit: Update date to either 26th or 27th

## OS Changes
- Add default bind directories for ephemeral storage ([#632])

> - Add support for `net.toml` under `/usr/share/bottlerocket` for all variants ([#524])

Reviewer: Missing attribution for pb8o@

> - Add support for `net.toml` under `/usr/share/bottlerocket` for all variants ([#524]) - Thanks @pb80

Reviewer 2: ```suggestion
- Extend netdog to look for `net.toml` under `/usr/share/bottlerocket` ([#524]) - Thanks @pb80
```

- Add `containerd-2.1` package ([#621])
- Add `systemd-257` package ([#581])

> - Add setting to control `static-pods` enablement ([#641])

Reviewer 2: This should go to kubernetes. 

```
- Add support for Kubernetes Setting `static-pods-enabled`
```

> - Update `host-ctr` aws-sdk-go to v2 and compile with go 1.24 ([#642])

Reviewer 2: ```suggestion
- Update `host-ctr` to migrate to `aws-sdk-go-v2` and bump to go 1.24 ([#642])
```

## Orchestrator Changes
### Kubernetes

> - Enable `MutableCSINodeAllocatableCount` in kubernetes-1.34 ([#634])

Reviewer: Missing attribution for torredil@

> - Enable `MutableCSINodeAllocatableCount` in kubernetes-1.34 ([#634]) - Thanks @torredil

Reviewer 2: ```suggestion
- Enable `MutableCSINodeAllocatableCount` feature gate on kubelet for kubernetes-1.34 ([#634]) - Thanks @torredil
```
